### PR TITLE
Use N or N-1 instead of empirical kmax in degree histogram

### PR DESCRIPTION
**Pull request by sdmccabe**
Tweaks `DegreeDivergence` to use N (if there are self-loops) or N-1 (if there
are not) instead of the largest observed degree when constructing the degree
histogram. This works because a `Counter` behaves like a `defaultdict(int)`, so
the list comprehension

```
hist = np.array([counter[v] for v in range(max_deg)])
```

pads out the zeros automatically. It should rarely affect results, but has the
advantages of (i) being slightly more accurate for the pedants among us and (ii)
being faster for some reason. And while it doesn't affect the JSD, if we ever do
implement #174, maybe it ends up mattering somewhere else.

### CODE REVIEW

**Comment by leotrs** [comment]
Wait. So when there aren't any self-loops, the sum of all degrees is 2m. So the normalization constant in that case should be 2m so that the sum of all the bins is 1. Right?

Can't we just divide by the total sum at the end and cover all cases?

**Comment by sdmccabe** [comment]
Nothing in this code normalizes. The normalization is passed to `js_divergence`. This is just about constructing the histogram that is normalized. The previous (just fine) code said that the histogram included degrees between zero and the _empirical_ kmax, instead of between zero and the _theoretical_ kmax for the graph (thus the N/N-1) distinction. In practice this doesn't matter because the zeros in the histogram between kmax+1 and N don't contribute to the JSD, but technically they should be included, and it might matter (but probably won't) for other measures. 

Basically, this is all incredibly low stakes.

**Comment by leotrs** [comment]
Oooooooh ok I get it now, sorry for the confusion. So the question is whether we want to return a histogram with a bunch of zeros at the end? Memory-wise the current version is better, but the one you propose is faster (according to you). So I'm just going to approve this to forget about it until it comes biting us back.
